GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/GVFSUpgradeReminderTests.cs

```diff
@@ -62,6 +63,51 @@ public void NoReminderForLeftOverDownloads()
             this.VerifyUpgradeVerbStopsReminder();
         }
 
+        [TestCase]
+        public void UpgradeTimerScheduledOnServiceStart()
+        {
+            this.RestartService();
+            Thread.Sleep(TimeSpan.FromSeconds(15));
+
+            this.ServiceLogContainsUpgradeMessaging().ShouldBeTrue();
+        }
+
+        private bool ServiceLogContainsUpgradeMessaging()
+        {
+            string upgradeTimerMessage = "Checking for product upgrades. (Start)";
+            string serviceLogFolder = Path.Combine(
+                Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData),
+                "GVFS",
+                GVFSServiceProcess.TestServiceName,
+                "Logs");
+            DirectoryInfo logsDirectory = new DirectoryInfo(serviceLogFolder);
+            IEnumerable<FileInfo> files = logsDirectory.GetFiles()
+                .OrderByDescending(f => f.LastWriteTime)
+                .Take(2);

```

Comment:
Why take 2?

Reply:
that was by mistake. Corrected. Using `FirstOrDefault()` now.